cmd/cni-metrics-helper/metrics/pod_watcher.go

```diff
@@ -33,9 +32,15 @@ func NewDefaultPodWatcher(k8sClient client.Client, log logger.Logger) *defaultPo
 func (d *defaultPodWatcher) GetCNIPods(ctx context.Context) ([]string, error) {
 	var CNIPods []string
 	var podList corev1.PodList
+	labelSelector, _ := metav1.LabelSelectorAsSelector(&metav1.LabelSelector{
```

Comment:
You're ignoring errors :)
Since the input here is all static, I think the only way this fails is if we make a typo in the labelselector somewhere (ie: a programmer error).  So I think it's reasonable to `panic()` if this returns an error.

We should _not_ ignore the error and continue, because we don't know what state the labelSelector might be in.  Eg: it might be partially constructed and not match any pods (or might have a dangling null and we crash).

Reply:
Sure, Will add a check.

Reply:
Right.  It's ~easy to verify the code is correct now - but I'm thinking about what it looks like after patches by 20 different people who didn't realise these assumptions.

Fail loudly and obviously at compile time == best; fail loudly and obviously at every execution == next best; [.. other options]; fail by ignoring errors and silently doing the wrong thing sometimes at runtime == worst.